elasticsearch_acls.go

```diff
@@ -0,0 +1,250 @@
+	ElasticsearchAclPermissionWrite        = "write"
+	ElasticsearchAclPermissionAdmin        = "admin"
+	ElasticsearchAclPermissionDeny         = "deny"
+	ElasticsearchAclPermissionReadAndWrite = "readwrite"
+)
+
+// list of supported Aiven API Elasticsearch ACL permissions
+var elasticsearchPermissins = []string{
+	ElasticsearchAclPermissionRead,
+	ElasticsearchAclPermissionWrite,
+	ElasticsearchAclPermissionAdmin,
+	ElasticsearchAclPermissionDeny,
+	ElasticsearchAclPermissionReadAndWrite,
+}
+
+type (
+	// ElasticSearchACLsHandler Aiven go-client handler for Elastisearch ACLs
+	ElasticSearchACLsHandler struct {
+		client *Client
+	}
+
+	// ElasticsearchACLRequest Aiven API request
+	// https://api.aiven.io/v1/project/<project>/service/<service_name>/elasticsearch/acl
+	ElasticsearchACLRequest struct {
+		ElasticSearchACLConfig ElasticSearchACLConfig `json:"elasticsearch_acl_config"`
+	}
+
+	// ElasticSearchACLResponse Aiven API response
+	// https://api.aiven.io/v1/project/<project>/service/<service_name>/elasticsearch/acl
+	ElasticSearchACLResponse struct {
+		APIResponse
+		ElasticSearchACLConfig ElasticSearchACLConfig `json:"elasticsearch_acl_config"`
+	}
+
+	// ElasticSearchACLConfig represents a configuration for Elasticsearch ACLs
+	ElasticSearchACLConfig struct {
+		ACLs        []ElasticSearchACL `json:"acls"`
+		Enabled     bool               `json:"enabled"`
+		ExtendedAcl bool               `json:"extendedAcl"`
+	}
+
+	// ElasticSearchACL represents a ElasticSearch ACLs entry
+	ElasticSearchACL struct {
+		Rules    []ElasticsearchACLRule `json:"rules"`
+		Username string                 `json:"username"`
+	}
+
+	// ElasticsearchACLRule represents a ElasticSearch ACLs Rule entry
+	ElasticsearchACLRule struct {
+		Index      string `json:"index"`
+		Permission string `json:"permission"`
+	}
+)
+
+// Create creates new Elasticsearch ACLs config
+func (h *ElasticSearchACLsHandler) Create(project, service string, req ElasticsearchACLRequest) (*ElasticSearchACLResponse, error) {
+	err := req.ElasticSearchACLConfig.Check()
+	if err != nil {
+		return nil, err
+	}
+
+	// POST or PUT request
+	path := buildPath("project", project, "service", service, "elasticsearch", "acl")
+	bts, err := h.client.doPostRequest(path, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return h.response(bts)
+}
+
+// Update updates Elasticsearch ACL config
+func (h *ElasticSearchACLsHandler) Update(project, service string, req ElasticsearchACLRequest) (*ElasticSearchACLResponse, error) {
+	err := req.ElasticSearchACLConfig.Check()
+	if err != nil {
+		return nil, err
+	}
+
+	path := buildPath("project", project, "service", service, "elasticsearch", "acl")
+	bts, err := h.client.doPutRequest(path, req)
+	if err != nil {
+		return nil, err
+	}
+
+	return h.response(bts)
+}
+
+// List lists all existing Elasticsearch ACLs config
+func (h *ElasticSearchACLsHandler) List(project, service string) (*ElasticSearchACLResponse, error) {
```

Comment:
I still think this should be Get as there is only one config for each service

Reply:
ok, I will change that